project1-sps/script.js

```diff
@@ -1,4 +1,92 @@
+// initial mode & username variable
+var currentMode = "Enter Name";
+var userName = ""; // empty string, nothing stored
+
+// record user's Win Loss Draw score
+var userWinRecord = 0;
+var userLoseRecord = 0;
+var userDrawRecord = 0;
+
+// computerOption helper function
+var computerSPS = function (input) {
+  var randomNumber = Math.floor(Math.random() * 3) + 1;
+  console.log(randomNumber);
+  if (randomNumber == 1) {
+    return "scissors";
+  } else if (randomNumber == 2) {
+    return "paper";
+  } else {
+    return "stone";
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue;
+  // current game mode
+  if (currentMode === "Enter Name") {
+    userName = input;
+    // change to start game mode
+    currentMode = "Start Game";
+    return `Hi ${userName}! Please enter game input - scissors, paper, stone`;
+  } else if (currentMode === "Start Game") {
+    // user input validation
+    // if user input not equals to scissors/paper/stone, run invalid msg
+    if (!input == "scissors" || "paper" || "stone") {
+      var myOutputValue =
+        "Please enter a valid game input - scissors, paper, stone";
+    }
```

Comment:
Nice input validation here! Since we are defining `myOutputValue` before this, line 35 should be a reassignment - no need to repeat the keyword var